src/Http/Routing/test/UnitTests/Builder/MinimalActionEndpointRouteBuilderExtensionsTest.cs

```diff
@@ -359,6 +359,84 @@ public void MapFallbackWithoutPath_BuildsEndpointWithLowestRouteOrder()
             Assert.Equal(int.MaxValue, routeEndpointBuilder.Order);
         }
 
+        [Fact]
+        // This test scenario simulates methods defined in a top-level program
+        // which are compiler generated. We currently do some manually parsing leveraging
+        // code in Roslyn to support this scenario. More info at https://github.com/dotnet/roslyn/issues/55651.
+        public void MapMethod_DoesNotEndpointNameForInnerMethod()
+        {
+            var name = "InnerGetString";
+            var builder = new DefaultEndpointRouteBuilder(new ApplicationBuilder(new EmptyServiceProvdier()));
+            string InnerGetString() => "TestString";
```

Comment:
```suggestion
            var testString = "TestString";
            string InnerGetString() => testString;
```

This breaks the test. I think it's because the `MethodInfo` in this case `IsCompilerGeneratedMethod` but it has a non-null `Target`.

Reply:
Good catch! I think the `action.Target == null` isn't really needed now that the `IsCompilerGeneratedMethod` check does the work of walking up the declaring types to check to see if a method is declared. I was incorrectly using `action.Target == null` to check for local functions.